plugins/hydra_ax_sweeper/tests/test_ax_sweeper_plugin.py

```diff
@@ -191,39 +192,128 @@ def test_configuration_set_via_cmd_and_default_config(
         assert "quadratic_y" in best_parameters
 
 
-def test_ax_logging(tmpdir: Path) -> None:
+@pytest.mark.parametrize(  # type: ignore
+    "cmd_arg, expected_str",
+    [
+        ("polynomial.y=choice(-1, 0, 1)", "polynomial.y: choice=[-1, 0, 1]"),
+        ("polynomial.y=range(-1, 2)", "polynomial.y: choice=[-1, 0, 1]"),
+        ("polynomial.y=range(-1, 3, 1)", "polynomial.y: choice=[-1, 0, 1, 2]"),
+        (
+            "polynomial.y=range(-1, 2, 0.5)",
+            "polynomial.y: choice=[-1.0, -0.5, 0.0, 0.5, 1.0, 1.5]",
+        ),
+        ("polynomial.y=int(interval(-1, 2))", "polynomial.y: range=[-1, 2]"),
+        ("polynomial.y=interval(-1, 2)", "polynomial.y: range=[-1.0, 2.0]"),
+        ("polynomial.y=2", "polynomial.y: fixed=2"),
+        ("polynomial.y=2.0", "polynomial.y: fixed=2.0"),
+    ],
+)
+def test_ax_logging(tmpdir: Path, cmd_arg: str, expected_str: str) -> None:
     cmd = [
-        sys.executable,
         "tests/apps/polynomial.py",
         "-m",
         "hydra.run.dir=" + str(tmpdir),
-        "polynomial.x=-5:-2",
-        "polynomial.y=-1,1",
+        "polynomial.x=interval(-5, -2)",
         "polynomial.z=10",
         "hydra.sweeper.ax_config.max_trials=2",
-    ]
-    result = subprocess.check_output(cmd).decode("utf-8").rstrip()
-    assert "polynomial.x: range=[-5, -2], type = int" in result
-    assert "polynomial.y: choice=[-1, 1], type = int" in result
-    assert "polynomial.z: fixed=10, type = int" in result
+    ] + [cmd_arg]
+    result, _ = get_run_output(cmd)
+    assert "polynomial.x: range=[-5.0, -2.0]" in result
+    assert expected_str in result
+    assert "polynomial.z: fixed=10" in result
+
+
+@pytest.mark.parametrize(  # type: ignore
+    "cmd_arg, serialized_encoding, best_coefficients, best_value",
+    [
+        (
+            "polynomial.coefficients=[-1, 0, 1],[2, 3, 4],[5, 6, 7]",
+            "choice=['[-1,0,1]', '[2,3,4]', '[5,6,7]']",
+            "'[-1,0,1]'",
+            101.0,
+        ),
+        (
+            "polynomial.coefficients=choice([8, 12, 11],[-1, -1, 1000], [-2, 4, 7])",
+            "choice=['[8,12,11]', '[-1,-1,1000]', '[-2,4,7]']",
+            "'[-2,4,7]'",
+            447,
+        ),
+    ],
+)
+def test_jobs_using_choice_between_lists(
+    tmpdir: Path,
+    cmd_arg: str,
+    serialized_encoding: str,
+    best_coefficients: str,
+    best_value: float,
+) -> None:
+    cmd = [
+        "tests/apps/polynomial_with_list_coefficients.py",
+        "-m",
+        "hydra.run.dir=" + str(tmpdir),
+        "hydra.sweeper.ax_config.max_trials=3",
+    ] + [cmd_arg]
+    result, _ = get_run_output(cmd)
+    assert f"polynomial.coefficients: {serialized_encoding}" in result
+    assert f"'polynomial.coefficients': {best_coefficients}" in result
+    assert f"New best value: {best_value}" in result
+
+
+@pytest.mark.xfail  # type: ignore
```

Comment:
what is this expected to fail?

Reply:
This is the issue https://github.com/facebookresearch/hydra/issues/946. Seems it has been fixed. Will try without marking the test as failed.